torchmetrics/functional/regression/tweedie_deviance.py

```diff
@@ -0,0 +1,147 @@
+
+        term_1 = torch.pow(torch.max(targets, zero_tensor), 2 - power) / ((1 - power) * (2 - power))
+        term_2 = targets * torch.pow(preds, 1 - power) / (1 - power)
+        term_3 = torch.pow(preds, 2 - power) / (2 - power)
+        deviance_score = 2 * (term_1 - term_2 + term_3)
+
+    sum_deviance_score = torch.sum(deviance_score)
+    num_observations = torch.tensor(torch.numel(deviance_score))
+
+    return sum_deviance_score, num_observations
+
+
+def _tweedie_deviance_score_compute(sum_deviance_score: Tensor, num_observations: Tensor) -> Tensor:
+    """Computes Deviance Score.
+
+    Args:
+        sum_deviance_score: Sum of deviance scores accumalated until now.
+        num_observations: Number of observations encountered until now.
+
+    Example:
+        >>> targets = torch.tensor([1.0, 2.0, 3.0, 4.0])
+        >>> preds = torch.tensor([4.0, 3.0, 2.0, 1.0])
+        >>> sum_deviance_score, num_observations = _tweedie_deviance_score_update(preds, targets, power=0)
+        >>> _tweedie_deviance_score_compute(sum_deviance_score, num_observations)
+        tensor(5.)
+    """
+
+    return sum_deviance_score / num_observations
+
+
+def tweedie_deviance_score(preds: Tensor, targets: Tensor, power: float = 0.0) -> Tensor:
+    r"""
+    Computes the `Deviance Score <https://en.wikipedia.org/wiki/Tweedie_distribution#The_Tweedie_deviance>`_ between
+    targets and predictions:
+
+    .. math::
+        deviance\_score(\hat{y},y) =
+        \begin{cases}
+        (\hat{y} - y)^2, & \text{for }power=0\\
+        2 * (y * log(\frac{y}{\hat{y}}) + \hat{y} - y),  & \text{for }power=1\\
+        2 * (log(\frac{\hat{y}}{y}) + \frac{y}{\hat{y}} - 1),  & \text{for }power=2\\
+        2 * (\frac{(max(y,0))^{2}}{(1 - power)(2 - power)} - \frac{y(\hat{y})^{1 - power}}{1 - power} + \frac{(\hat{y})
+            ^{2 - power}}{2 - power}), & \text{otherwise}
+        \end{cases}
+
+    where :math:`y` is a tensor of targets values, and :math:`\hat{y}` is a tensor of predictions.
+
+    Args:
+        preds: Predicted tensor with shape ``(N,d)``
+        targets: Ground truth tensor with shape ``(N,d)``
+        power:
+            - power < 0 : Extreme stable distribution. (Requires: preds > 0.)
+            - power = 0 : Normal distribution. (Requires: targets and preds can be any real numbers.)
+            - power = 1 : Poisson distribution. (Requires: targets >= 0 and y_pred > 0.)
+            - 1 < p < 2 : Compound Poisson distribution. (Requires: targets >= 0 and preds > 0.)
+            - power = 2 : Gamma distribution. (Requires: targets > 0 and preds > 0.)
+            - power = 3 : Inverse Gaussian distribution. (Requires: targets > 0 and preds > 0.)
+            - otherwise : Positive stable distribution. (Requires: targets > 0 and preds > 0.)
+
+    Example:
+        >>> from torchmetrics.functional.regression import tweedie_deviance_score
```

Comment:
this is the recommended user import path
```suggestion
        >>> from torchmetrics.functional import tweedie_deviance_score
```